File: modules/day-02-reusable/python/starter.py
plants = [
    {"treatment": "control", "height_cm": 12.1, "leaf_count": 8},
    {"treatment": "control", "height_cm": 11.8, "leaf_count": 9},
    {"treatment": "fertilizer", "height_cm": 15.2, "leaf_count": 14},
    {"treatment": "fertilizer", "height_cm": 14.9, "leaf_count": 13},
    {"treatment": "shade", "height_cm": 10.5, "leaf_count": 6},
    {"treatment": "shade", "height_cm": 10.8, "leaf_count": 7},
]

control_height_values = [
    row["height_cm"] for row in plants if row["treatment"] == "control"
]
control_height_mean = sum(control_height_values) / len(control_height_values)
control_height_summary = f"control average height = {control_height_mean:.1f} cm"

fertilizer_height_values = [
    row["height_cm"] for row in plants if row["treatment"] == "fertilizer"
]
fertilizer_height_mean = sum(fertilizer_height_values) / len(fertilizer_height_values)
fertilizer_height_summary = f"fertilizer average height = {fertilizer_height_mean:.1f} cm"

shade_height_values = [row["height_cm"] for row in plants if row["treatment"] == "shade"]
shade_height_mean = sum(shade_height_values) / len(shade_height_values)
shade_height_summary = f"shade average height = {shade_height_mean:.1f} cm"

control_leaf_values = [row["leaf_count"] for row in plants if row["treatment"] == "control"]
control_leaf_mean = sum(control_leaf_values) / len(control_leaf_values)
control_leaf_summary = f"control average leaves = {control_leaf_mean:.1f}"

fertilizer_leaf_values = [
    row["leaf_count"] for row in plants if row["treatment"] == "fertilizer"
]
fertilizer_leaf_mean = sum(fertilizer_leaf_values) / len(fertilizer_leaf_values)
fertilizer_leaf_summary = f"fertilizer average leaves = {fertilizer_leaf_mean:.1f}"

shade_leaf_values = [row["leaf_count"] for row in plants if row["treatment"] == "shade"]
shade_leaf_mean = sum(shade_leaf_values) / len(shade_leaf_values)
shade_leaf_summary = f"shade average leaves = {shade_leaf_mean:.1f}"

print("Height summaries")
print(control_height_summary)
print(fertilizer_height_summary)
print(shade_height_summary)

print("Leaf summaries")
print(control_leaf_summary)
print(fertilizer_leaf_summary)
print(shade_leaf_summary)

File: modules/day-02-reusable/r/starter.R
plants <- data.frame(
  treatment = c("control", "control", "fertilizer", "fertilizer", "shade", "shade"),
  height_cm = c(12.1, 11.8, 15.2, 14.9, 10.5, 10.8),
  leaf_count = c(8, 9, 14, 13, 6, 7)
)

control_height_values <- plants$height_cm[plants$treatment == "control"]
control_height_mean <- mean(control_height_values)
control_height_summary <- paste(
  "control average height =",
  round(control_height_mean, 1),
  "cm"
)

fertilizer_height_values <- plants$height_cm[plants$treatment == "fertilizer"]
fertilizer_height_mean <- mean(fertilizer_height_values)
fertilizer_height_summary <- paste(
  "fertilizer average height =",
  round(fertilizer_height_mean, 1),
  "cm"
)

shade_height_values <- plants$height_cm[plants$treatment == "shade"]
shade_height_mean <- mean(shade_height_values)
shade_height_summary <- paste(
  "shade average height =",
  round(shade_height_mean, 1),
  "cm"
)

control_leaf_values <- plants$leaf_count[plants$treatment == "control"]
control_leaf_mean <- mean(control_leaf_values)
control_leaf_summary <- paste(
  "control average leaves =",
  round(control_leaf_mean, 1)
)

fertilizer_leaf_values <- plants$leaf_count[plants$treatment == "fertilizer"]
fertilizer_leaf_mean <- mean(fertilizer_leaf_values)
fertilizer_leaf_summary <- paste(
  "fertilizer average leaves =",
  round(fertilizer_leaf_mean, 1)
)

shade_leaf_values <- plants$leaf_count[plants$treatment == "shade"]
shade_leaf_mean <- mean(shade_leaf_values)
shade_leaf_summary <- paste(
  "shade average leaves =",
  round(shade_leaf_mean, 1)
)

print("Height summaries")
print(control_height_summary)
print(fertilizer_height_summary)
print(shade_height_summary)

print("Leaf summaries")
print(control_leaf_summary)
print(fertilizer_leaf_summary)
print(shade_leaf_summary)
